env0/utils.go

```diff
@@ -0,0 +1,141 @@
+package env0
+
+import (
+	"errors"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+var matchAllCap = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+// https://stackoverflow.com/questions/56616196/how-to-convert-camel-case-string-to-snake-case
+func toSnakeCase(str string) string {
+	snake := matchFirstCap.ReplaceAllString(str, "${1}_${2}")
+	snake = matchAllCap.ReplaceAllString(snake, "${1}_${2}")
+	return strings.ToLower(snake)
+}
+
+func boolPtr(b bool) *bool {
+	return &b
+}
+
+func stringPtr(s string) *string {
+	return &s
+}
+
+func intPtr(i int) *int {
+	return &i
+}
+
+// Extracts values from the resourcedata, and writes it to the interface.
+func readResourceData(i interface{}, d *schema.ResourceData) error {
+	val := reflect.ValueOf(i).Elem()
+	for i := 0; i < val.NumField(); i++ {
+		fieldName := val.Type().Field(i).Name
+		// Assumes golang is CamalCase and Terraform is snake_case.
+		fieldNameSC := toSnakeCase(fieldName)
+		field := val.Field(i)
+		fieldType := field.Type()
+
+		dval, ok := d.GetOk(fieldNameSC)
+		if !ok {
+			continue
+		}
```

Comment:
Don't we miss here on errors for "missed" attributes?

Reply:
When you're saying - "missed attribute": as in an attribute configured on the resource data that was not "injected" to the struct?

Reply:
@TomerHeber yes, exactly
I think we'd wanna have such error thrown, don't we? we'll get that in dev phase and make sure to adjust accordingly

Reply:
@yaronya :
https://github.com/env0/terraform-provider-env0/pull/279/commits/24ae56c8877b943682984a3aae20acb9df5ce8e7